backend/server.hs

```diff
@@ -34,24 +33,29 @@ removeClient client = filter ((/= fst client) . fst)
 
 sendDirectMessage :: Text -> Text -> ServerState -> IO ()
 sendDirectMessage user msg clients = do
-  let recipient = (words (T.unpack msg))!!1
-  let client = find (\(name, _)->(T.unpack name)==recipient) clients
+  let recipient = words (T.unpack msg)!!1
+  let client = find (\(name, _)->T.unpack name==recipient) clients
 
   case client of
-    Nothing -> broadcast ("SYSTEM: " `mappend` user `mappend` " is extremely stupid. She or he wanna send message to nobody") clients
-    Just (_, conn) -> WS.sendTextData conn (user `mappend` " directly to you: " `mappend` T.pack(drop (9+ (length $ T.unpack user)) (T.unpack msg)))
-
-
-
-
+    Nothing -> broadcast
+      ("SYSTEM: " `mappend` user `mappend` " is extremely stupid. She or he wanna send message to user that does not exist😳") clients
+    Just (_, conn) ->
+      WS.sendTextData conn (user `mappend` " directly to you: " `mappend` T.pack(drop  (8 + length (T.unpack user)) (T.unpack msg)))
 
+session :: WS.Connection -> MVar ServerState -> Client -> IO()
 session conn state client = do
    modifyMVar_ state $ \s -> do
      let s' = addClient client s
      broadcast ("SYSTEM: " `mappend` fst client `mappend` " joined.") s'
      return s'
    talk conn state client
 
+
+sendMessageWithVulgarism :: MVar ServerState -> Text -> Text -> IO()
+sendMessageWithVulgarism state user msg =
+  readMVar state >>= broadcast
+    (user `mappend`": " `mappend` T.replace "fuck" "f*ck" msg)
```

Comment:
Small range of vulgarisms filtered by the function. That's not a point of this project to make a complex product, I am aware, but it could be demonstrated easily that it is possible to replace much more than one vulgar word.